src/app/hooks/useDebounce.ts

```diff
@@ -0,0 +1,19 @@
+import { useState, useEffect } from "react";
+
+function useDebounce(value:string, delay:number) {
```

Comment:
Does it always have to be a string? Could work with other values theoretically.
I guess the other way to do it is to pass a function rather than a string. More like lodash https://lodash.com/docs/4.17.15#debounce

Reply:
@matt-greenfield-15 changed type from string to T